test/unit/gateways/sum_up_test.rb

```diff
@@ -0,0 +1,361 @@
+require 'test_helper'
+
+class SumUpTest < Test::Unit::TestCase
+  def setup
+    @gateway = SumUpGateway.new(
+      token: 'sup_sk_ABC123',
+      pay_to_email: 'example@example.com'
+    )
+    @credit_card = credit_card
+    @amount = 100
+
+    @options = {
+      payment_type: 'card',
+      billing_address: address,
+      description: 'Store Purchase'
+    }
+  end
+
+  def test_successful_purchase
+    @gateway.expects(:ssl_request).returns(successful_create_checkout_response)
+    @gateway.expects(:ssl_request).returns(successful_complete_checkout_response)
+    response = @gateway.purchase(@amount, @credit_card, @options)
+
+    assert_success response
+
+    assert_equal 'PENDING', response.message
+    assert response.test?
+  end
+
+  def test_failed_purchase
+    @credit_card.year = 2020
+    @gateway.expects(:ssl_request).returns(failed_complete_checkout_response)
+    response = @gateway.purchase(@amount, @credit_card, @options)
+
+    assert_failure response
+    assert_equal SumUpGateway::STANDARD_ERROR_CODE_MAPPING[:multiple_invalid_parameters], response.error_code
+  end
+
+  def test_scrub
+    assert @gateway.supports_scrubbing?
+    assert_equal @gateway.scrub(pre_scrubbed), post_scrubbed
+  end
```

Comment:
💬 Comment:

Maybe so small test for some private methods like would ensure that adapter behavior is declared.

- success_from
- message_from
- authorization_from
- format_multiple_errors
- error_code_from